### PR TITLE
Adds ttl related members/methods to the WriteRequest class.

### PR DESCRIPTION
**Pull request by mehmetg**
Fixes typo in method name.
Bumps version for release v1.1.7

### CODE REVIEW

File: satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java [hunk 1]
```diff
@@ -20,6 +22,8 @@
   private String channel;
   private T message;
   private String position;
+  private Long ttl;
+  private T ttl_message;
```
Comment (khabinov) [inline satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java]: Camel case? Or, if we want it to match RTM spec, then might be better to use annotations.

File: satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java [hunk 2]
```diff
@@ -44,4 +58,27 @@ public T getMessage() {
   public String getPosition() {
     return position;
   }
+
+  public long getTtl() {
```
Comment (khabinov) [inline satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java]: type safety question: what if ttl is null?

File: satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java [hunk 3]
```diff
@@ -33,6 +37,16 @@ public WriteRequest(String channel, T message, String position) {
     this.position = position;
   }
 
+
+  public WriteRequest(String channel, T message, String position, final long ttl, final T ttl_message) {
+    Preconditions.checkArgument(ttl > 0, "ttl must be non negative");
```
Comment (khabinov) [inline satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java]: Makes sense to add ttl value to error message.

File: satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java [hunk 4]
```diff
@@ -33,6 +37,16 @@ public WriteRequest(String channel, T message, String position) {
     this.position = position;
   }
 
+
+  public WriteRequest(String channel, T message, String position, final long ttl, final T ttl_message) {
```
Comment (khabinov) [inline satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java]: Should `PublishRequest` be updated too?

File: satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java (hunk 1 above)
Comment (khabinov) [reply satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java]: I have some doubts whether `message` and `ttl_message` are of the same type. Should we have a separate generic type for it?

Comment (mehmetg) [reply satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java]: The casing is the way it is to support both gson and jackson. IIRC snake case is required for jackson.

As for the `message` and `ttl_message`, yes it is a fair concern I'll add a TODO for later.

File: satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java (hunk 4 above)
Comment (mehmetg) [reply satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java]: Added.

File: satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java (hunk 3 above)
Comment (mehmetg) [reply satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java]: Done.

File: satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java (hunk 2 above)
Comment (mehmetg) [reply satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java]: Done.

File: satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java [hunk 5]
```diff
@@ -33,6 +38,17 @@ public WriteRequest(String channel, T message, String position) {
     this.position = position;
   }
 
+
+
+  public WriteRequest(String channel, T message, String position, final long ttl, final T ttl_message) {
+    Preconditions.checkArgument(ttl > 0, String.format("ttl must be non negative: %s", ttl));
```
Comment (khabinov) [inline satori-rtm-sdk-core/src/main/java/com/satori/rtm/model/WriteRequest.java]: Btw, no need for `String.format`, https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Preconditions.html#checkArgument(boolean, java.lang.String, java.lang.Object...)

**Pull request merged**